quickwit/quickwit-search/src/root.rs

```diff
@@ -389,6 +402,106 @@ pub async fn root_search(
     })
 }
 
+pub(crate) struct ExtractTimestampRange<'a> {
```

Comment:
I think having a function 

```pub(crate) fn refine_start_end_timestamp_from_ast(query_ast: &QueryAst, start_timestamp: &mut Option<i64>, end_timestamp: &mut Option<i64>)```

Would have been beneficial here...
ExtractTimestampRange could then have been private... And possibly in the query ast files.